AppCenterAnalytics/AppCenterAnalytics/Internals/Model/MSEventLog.m

```diff
@@ -68,53 +82,114 @@ - (MSCommonSchemaLog *)toCommonSchemaLogForTargetToken:(NSString *)token {
   // Event name goes to part A.
   csLog.name = self.name;
 
+  // Metadata extension must accompany data.
   // Event properties goes to part C.
-  MSCSData *data = [MSCSData new];
-  csLog.data = data;
-  csLog.data.properties = [self convertACPropertiesToCSproperties:self.properties];
+  csLog.data = [MSCSData new];
+  csLog.ext.metadataExt = [MSMetadataExtension new];
+  [self setPropertiesAndMetadataForCSLog:csLog];
   return csLog;
 }
 
 #pragma mark - Helper
 
-- (NSDictionary<NSString *, NSObject *> *)convertACPropertiesToCSproperties:(NSDictionary<NSString *, NSString *> *)acProperties {
+- (void)setPropertiesAndMetadataForCSLog:(MSCommonSchemaLog *)csLog {
   NSMutableDictionary *csProperties;
-  if (acProperties) {
+  NSMutableDictionary *metadata;
+  if (self.typedProperties) {
     csProperties = [NSMutableDictionary new];
-    for (NSString *acKey in acProperties) {
+    metadata = [NSMutableDictionary new];
+    for (MSTypedProperty *typedProperty in [self.typedProperties.properties objectEnumerator]) {
 
       // Properties keys are mixed up with other keys from Data, make sure they don't conflict.
-      if ([acKey isEqualToString:kMSDataBaseData] || [acKey isEqualToString:kMSDataBaseDataType]) {
-        MSLogWarning(MSAnalytics.logTag, @"Cannot use %@ in properties, skipping that property.", acKey);
+      if ([typedProperty.name isEqualToString:kMSDataBaseData] || [typedProperty.name isEqualToString:kMSDataBaseDataType]) {
+        MSLogWarning(MSAnalytics.logTag, @"Cannot use %@ in properties, skipping that property.", typedProperty.name);
         continue;
       }
+      [self addTypedProperty:typedProperty toCSMetadata:metadata AndCSProperties:csProperties];
```

Comment:
Any reason to use dictionary instead of json object?

Reply:
For what object? None of the models introduce Json at this level, and property conversion was already using dictionary, so we wanted to be consistent.

Is there a very strong reason to rewrite it that way?

Reply:
I got confused with Android. Android has a JSON object instead of map (dictionary). Never mind.